cli/config.ts

```diff
@@ -0,0 +1,46 @@
+import { exists, mkdir, readFile, writeFile } from "fs";
+import { homedir } from "os";
+import { join } from "path";
+import { promisify } from "util";
+
+export interface IConfigSettings {
+  allowAnonymousAnalytics?: boolean;
+  userId?: string;
+}
+
+export function getConfigDirPath() {
+  return join(homedir(), ".dataform");
+}
+
+export function getConfigSettingsPath() {
+  return join(getConfigDirPath(), "settings.json");
+}
+
+export async function getConfigSettings(): Promise<IConfigSettings> {
+  try {
+    if (await promisify(exists)(getConfigSettingsPath())) {
+      return JSON.parse(
+        await promisify(readFile)(getConfigSettingsPath(), "utf8")
+      ) as IConfigSettings;
+    }
+  } catch (e) {
+    // If we can't get the config, just return the default.
+  }
+  return {};
```

Comment:
i realize that it's already effectively the default, but can you add an explicit `allowAnonymousAnalytics: false` here (more as documentation than anything else)?

Reply:
This would change the behaviour, there are 3 states here - false, true, undefined. I explicitly check for undefined so I can't set it here (so I know if to ask or not).

Reply:
OK, but doesn't that mean that users who are in some buggy state where the file can never be read (I don't know, weird environment or something), they keep being asked over and over?